app/src/ui/lib/filter-list.tsx

```diff
@@ -71,8 +72,20 @@ interface IFilterListProps<T extends IFilterListItem> {
   /** Called to render content before/above the filter and list. */
   readonly renderPreList?: () => JSX.Element | null
 
-  /** Called when an item is clicked. */
-  readonly onItemClick?: (item: T) => void
+  /**
+   * This function will be called when a pointer device is pressed and then
+   * released on a selectable row. Note that this follows the conventions
+   * of button elements such that pressing Enter or Space on a keyboard
+   * while focused on a particular row will also trigger this event. Consumers
+   * can differentiate between the two using the source parameter.
+   *
+   * Note that this event handler will not be called for keyboard events
+   * if event.preventDefault was called in the onRowKeyDown event handler.
```

Comment:
:art: should these mentions be `event.preventDefault()` (with the parentheses)?

Also, if you backtick these code segments some editors are nice enough to format them like code:

<img width="537"  src="https://user-images.githubusercontent.com/359239/50111251-c0462480-0212-11e9-9eb3-d987f5b8a92a.png">